compiler/rustc_driver/src/lib.rs

```diff
@@ -581,18 +590,24 @@ pub fn try_process_rlink(sess: &Session, compiler: &interface::Compiler) -> Comp
             sess.init_crate_types(collect_crate_types(sess, &[]));
             let outputs = compiler.build_output_filenames(sess, &[]);
             let rlink_data = fs::read(file).unwrap_or_else(|err| {
-                sess.fatal(&format!("failed to read rlink file: {}", err));
+                sess.fatal(RlinkUnableToRead {
+                    span: Default::default(),
```

Comment:
You don't need a `Span` if the original error didn't have one, just don't add a `span` field and don't annotate any fields with `#[primary_span]`.